week1/tuesday/Assignment_1_2_2/main.cpp

```diff
@@ -0,0 +1,82 @@
+#include <iostream>
+#include <string>
+
+bool has_punct(std::string pass)
+{
+    for (size_t i = 0; i < pass.size(); ++i)
+    {
+        if (std::ispunct(static_cast<unsigned char>(pass[i])))
+            return true;
+    }
+    return false;
+}
+
+bool has_digit(std::string pass)
+{
+    for (size_t i = 0; i < pass.size(); ++i)
+    {
+        if (std::isdigit(static_cast<unsigned char>(pass[i])))
+            return true;
+    }
+    return false;
+}
+
```

Comment:
static casting is unnecessary but exposes whats under the hood.
It is clear if you use the algorithm library for this.
eq. bool it = std::any_of(pass.begin(), pass.end(), [](unsigned char c){return std::isdigit(c);});